lib/tools/apk-signing.js

```diff
@@ -40,7 +40,9 @@ apkSigningMethods.executeApksigner = async function executeApksigner (args = [])
     ...args
   ];
   log.debug(`Starting apksigner: ${util.quote(fullCmd)}`);
-  const {stdout, stderr} = await exec(fullCmd[0], fullCmd.slice(1));
+  const {stdout, stderr} = await exec(fullCmd[0], fullCmd.slice(1), {
+    cwd: path.dirname(apkSignerJar)
```

Comment:
Please add a link to the original issue that explains why it is necessary to set cwd

Reply:
Ok, happy to; here goes:

It is necessary to set the cwd here because as discovered in https://github.com/appium/appium/issues/14724, somehow it is possible when appium has been running for a long time (like more than a week) for it to somehow lose it's CWD or something such that when launching `java` it doesn't know the cwd.  Being explicit here works around that issue, although I admit we still have the unanswered question of why does this happen :)

```
Cannot verify the signature of '/home/deployer/node_modules/appium/node_modules/appium-uiautomator2-server/apks/appium-uiautomator2-server-v4.15.0.apk'. Original error: Error occurred during initialization of VM
java.lang.Error: Properties init: Could not determine current working directory.
	at java.lang.System.initProperties(java.base/Native Method)
	at java.lang.System.initPhase1(java.base/System.java:1948)
```

Reply:
no need for such details. A simple link to the proper issue comment would suffice

Reply:
oh, sorry I did it wrong; I guess this is what you're looking for?:

https://github.com/appium/appium/issues/14724#issuecomment-737446715

Reply:
yes, that's it. Just add it as a comment to your patch

Reply:
Oh, I finally get what you're saying.  For some reason when you wrote this I thought you were talking about github comments:

> Please add a link to the original issue in comments that explains why it is necessary to set cwd

If you had instead written this, I probably wouldn't have gotten confused: 

> Please add a link to the original issue as a comment in the code that explains why it is necessary to set cwd

Anyway, sorry about the misunderstanding.  I added this commit to the branch, so I think we're good now, yes?  3fe1cf76addbcda00105afebda2557dde69db107